### ISSUE TITLE
feat: decoupled clipping (DAPO-style asymmetric bounds)

### ISSUE THREAD

**Issue by teilomillet** [teilomillet/textpolicy#6]
## Parent Issue
Part of #5 (GRPO optimizations for reasoning emergence)

## Problem

Current `policy_loss()` uses symmetric clipping:
```python
clipped_ratio = mx.clip(ratio, 1 - clip_ratio, 1 + clip_ratio)
```

This treats upward and downward probability changes equally, which can cause **entropy collapse** — the model becomes too confident too quickly, limiting exploration needed for reasoning emergence.

## Research Evidence

DAPO (ByteDance/Tsinghua, 2025) introduces **Clip-Higher**:
- Lower bound: `1 - ε_low` (e.g., 0.8)
- Upper bound: `1 + ε_high` (e.g., 1.28)

Asymmetric bounds allow the model to **increase** probabilities of good actions more easily than decreasing probabilities of bad actions. This promotes diversity and prevents entropy collapse.

> "Clip-Higher increases the upper clip range of the importance sampling ratio in policy gradient loss to promote diversity and prevent entropy collapse."

**Source:** [DAPO: An Open-Source LLM Reinforcement Learning System at Scale](https://arxiv.org/abs/2503.14476)

## Proposed Change

```python
def policy_loss(
    old_logprobs: mx.array,
    new_logprobs: mx.array,
    advantages: mx.array,
    clip_ratio_low: float = 0.2,   # Lower bound offset
    clip_ratio_high: float = 0.28  # Upper bound offset (asymmetric)
) -> mx.array:
    ratio = mx.exp(new_logprobs - old_logprobs)
    clipped_ratio = mx.clip(ratio, 1 - clip_ratio_low, 1 + clip_ratio_high)
    surr1 = ratio * advantages
    surr2 = clipped_ratio * advantages
    loss = -mx.mean(mx.minimum(surr1, surr2))
    return loss
```

## Impact

- Prevents entropy collapse during long training runs
- Maintains exploration capacity for Phase 2 (strategic planning)
- **Proven:** DAPO achieved 50 on AIME24 with 50% fewer steps than DeepSeek-R1-Zero

## Acceptance Criteria

- [ ] `policy_loss()` accepts separate `clip_ratio_low` and `clip_ratio_high` params
- [ ] Default values: low=0.2, high=0.28 (DAPO defaults)
- [ ] Backward compatible (single `clip_ratio` param still works)
- [ ] Unit tests for asymmetric clipping behavior
- [ ] Metrics track clip fraction for upper vs lower bounds separately
